streaming/src/main/scala/org/apache/spark/streaming/util/FileBasedWriteAheadLog.scala

```diff
@@ -47,7 +47,8 @@ private[streaming] class FileBasedWriteAheadLog(
     logDirectory: String,
     hadoopConf: Configuration,
     rollingIntervalSecs: Int,
-    maxFailures: Int
+    maxFailures: Int,
+    closeAfterWrite: Boolean = false
```

Comment:
Yeah, it is not exposed. So its better to actually keep it as non-default parameter, and add the parameter judiciously in the all the 4 spots.